miniapps/dpg/pmaxwell.cpp

```diff
@@ -0,0 +1,1444 @@
+                               std::vector<complex<double>> &curlcurlE);
+
+void source_function(const Vector &x, Vector & f);
+
+int dim;
+int dimc;
+double omega;
+double mu = 1.0;
+double epsilon = 1.0;
+
+enum prob_type
+{
+   plane_wave,
+   fichera_oven,
+   pml_general,
+   pml_plane_wave_scatter,
+   pml_pointsource
+};
+
+static const char *enum_str[] =
+{
+   "plane_wave",
+   "fichera_oven",
+   "pml_general",
+   "pml_plane_wave_scatter",
+   "pml_pointsource"
+};
+
+prob_type prob;
+
+int main(int argc, char *argv[])
+{
+   Mpi::Init();
+   int myid = Mpi::WorldRank();
+   Hypre::Init();
+
+   const char *mesh_file = "../../data/inline-quad.mesh";
+   int order = 1;
+   int delta_order = 1;
+   double rnum=1.0;
+   double theta = 0.0;
+   bool static_cond = false;
+   int iprob = 0;
+   int sr = 0;
+   int pr = 1;
+   bool exact_known = false;
+   bool with_pml = false;
+   bool visualization = true;
+   bool paraview = false;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree)");
+   args.AddOption(&rnum, "-rnum", "--number_of_wavelengths",
+                  "Number of wavelengths");
+   args.AddOption(&mu, "-mu", "--permeability",
+                  "Permeability of free space (or 1/(spring constant)).");
+   args.AddOption(&epsilon, "-eps", "--permittivity",
+                  "Permittivity of free space (or mass constant).");
+   args.AddOption(&iprob, "-prob", "--problem", "Problem case"
+                  " 0: plane wave, 1: Fichera 'oven', "
+                  " 2: Generic PML problem with point source given as a load "
+                  " 3: Scattering of a plane wave, "
+                  " 4: Point sourceg given on the boundary");
+   args.AddOption(&delta_order, "-do", "--delta_order",
+                  "Order enrichment for DPG test space.");
+   args.AddOption(&theta, "-theta", "--theta",
+                  "Theta parameter for AMR");
+   args.AddOption(&sr, "-sref", "--serial_ref",
+                  "Number of parallel refinements.");
+   args.AddOption(&pr, "-pref", "--parallel_ref",
+                  "Number of parallel refinements.");
+   args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
+                  "--no-static-condensation", "Enable static condensation.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&paraview, "-paraview", "--paraview", "-no-paraview",
+                  "--no-paraview",
+                  "Enable or disable ParaView visualization.");
+   args.Parse();
+   if (!args.Good())
+   {
+      if (myid == 0)
+      {
+         args.PrintUsage(cout);
+      }
+      return 1;
+   }
+   if (myid == 0)
+   {
+      args.PrintOptions(cout);
+   }
```

Comment:
You may want to move this print block after the `mesh_file` variable is reassigned below. The printed block of text shows the `inline-quad.mesh` even when this is overridden by the mesh-specific problems. There are different ways to handle this, such as leaving the default `mesh_file` empty, but moving the print statement has the advantage of displaying the file which will actually be read.